File: fabric/src/main/java/juuxel/adorn/lib/AdornEntitiesFabric.java
package juuxel.adorn.lib;

import juuxel.adorn.entity.SeatEntity;
import net.fabricmc.fabric.api.entity.event.v1.ServerPlayerEvents;

public final class AdornEntitiesFabric {
    public static void init() {
        ServerPlayerEvents.LEAVE.register(SeatEntity::stopSitting);
    }
}

File: fabric/src/main/kotlin/juuxel/adorn/lib/AdornEntitiesFabric.kt
package juuxel.adorn.lib

import juuxel.adorn.client.renderer.InvisibleEntityRenderer
import juuxel.adorn.entity.AdornEntities
import net.fabricmc.api.EnvType
import net.fabricmc.api.Environment
import net.fabricmc.fabric.api.client.rendering.v1.EntityRendererRegistry

object AdornEntitiesFabric {
    @Environment(EnvType.CLIENT)
    fun initClient() {
        EntityRendererRegistry.register(AdornEntities.SEAT, ::InvisibleEntityRenderer)
    }
}
